content/yaml-publishing/aws.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Amazon S3
+description: How to publish build artifacts to Amazon S3 using codemagic.yaml
+weight: 7
+aliases: [/publishing/publish-build-artifacts-to-amazon-s3, /flutter-publishing/publish-build-artifacts-to-amazon-s3, /knowledge-base/publish-build-artifacts-to-amazon-s3]
+---
+
+You can use custom scripts to publish your app artifacts to external sources. Here's an example for publishing to [Amazon S3](https://aws.amazon.com/s3/).
+
+1.  You will first need to provide Codemagic access to your Amazon S3 account for publishing. Add your credentials as [environment variables](..variables/environment-variable-groups) named `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` respectively to a group in the Codemagic UI which you can reference in your `codemagic.yaml` file. In the below example, we have created a group called `aws_credentials`. Make sure to select **Secure** when adding them.
+
+You can follow the [instructions](https://aws.amazon.com/getting-started/hands-on/backup-to-s3-cli/) provided by Amazon to create your account and get the necessary details.
+
+2.  Add the script to your `scripts` section before your build script to update the S3 bucket. `<FOLDER OR FILE>` refers to a specific folder or file to be synced. Replace `<BUCKET_NAME>` with your actual bucket name. Note that all the artifact files that Codemagic generates during the build are located in `FCI_BUILD_OUTPUT_DIR`.
+
+```yaml
+environment:
+    groups:
+      - aws_credentials
+  ... your other build steps
```

Comment:
Maybe rephrase it as "your other environment settings" or simply omit it
 > ... your other build steps